### PR TITLE
[FIX] base: Prevent traceback on rate creation

### PR DESCRIPTION
**Pull request by abz-odoo**
Steps:
 - Create new rate in random currency (for example USD)
 - Try to put 0 in USD per Unit
 - Try to put 0 in Unit per USD

Two errors are raised: ZeroDivisionError and Expected singleton

sentry-3916858495
sentry-3767286632

### CODE REVIEW

File: odoo/addons/base/models/res_currency.py [hunk 1]
```diff
@@ -361,7 +361,7 @@ def create(self, vals_list):
     def _get_latest_rate(self):
         # Make sure 'name' is defined when creating a new rate.
         if not self.name:
-            raise UserError(_("The date for the current rate is empty.\nPlease set it."))
+            raise UserError(_("The name for the current rate is empty.\nPlease set it."))
```
Comment (rigr-odoo) [inline odoo/addons/base/models/res_currency.py:364]: Should update the pot file too in stable

File: odoo/addons/base/models/res_currency.py [hunk 2]
```diff
@@ -400,11 +400,15 @@ def _inverse_company_rate(self):
     @api.depends('company_rate')
     def _compute_inverse_company_rate(self):
         for currency_rate in self:
+            if currency_rate.company_rate == 0:
+                currency_rate.company_rate = 1.0
```
Comment (rigr-odoo) [inline odoo/addons/base/models/res_currency.py:404]:
```suggestion
            if not currency_rate.company_rate:
                currency_rate.company_rate = 1.0
```

File: odoo/addons/base/models/res_currency.py [hunk 3]
```diff
@@ -400,11 +400,15 @@ def _inverse_company_rate(self):
...
             currency_rate.inverse_company_rate = 1.0 / currency_rate.company_rate
 
     @api.onchange('inverse_company_rate')
     def _inverse_inverse_company_rate(self):
         for currency_rate in self:
+            if currency_rate.inverse_company_rate == 0:
```
Comment (rigr-odoo) [inline odoo/addons/base/models/res_currency.py:410]: Same

**Comment by qdp-odoo** [comment]
@robodoo r+
@rigr-odoo good job spotting the missing .POT file :+1:

**Pull request closed without merging**